File: 946-validate-stack-sequences/946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> st;
        int cur = 0;
        for(int i=0;i<pushed.size();i++) {
            st.push(pushed[i]);
            while(!st.empty() && st.top()==popped[cur]) {
                st.pop();
                cur++;
            }
        }
        if(st.size()!=0)
            return false;
        return true;
    }
};

File: 946-validate-stack-sequences/946-validate-stack-sequences.java
class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> st = new Stack<>();
        int cur = 0; //pointer to iterate popped array
        for(int i=0;i<pushed.length;i++) {
            st.push(pushed[i]);
            while(!st.empty() && st.peek()==popped[cur]) {
                st.pop();
                cur++;
            }
                
        }
        if(st.empty())
            return true;
        return false;
    }
}